src/lib.rs

```diff
@@ -2048,13 +2056,14 @@ impl Connection {
             }
 
             // Create MAX_DATA frame as needed.
-            if self.should_update_max_data() {
+            if self.almost_full {
```

Comment:
If this is the signal that triggers sending the frame, you also need to replace the `self.should_update_max_data()` call in the `write_epoch()` method, otherwise it won't trigger sending a packet if there nothing else to send.

Reply:
Good point. Updated.